src/commcare_cloud/commands/ansible/helpers.py

```diff
@@ -64,28 +64,41 @@ def get_common_ssh_args(environment, use_factory_auth=False):
     return cmd_parts_with_common_ssh_args
 
 
-def get_default_ssh_options(environment, use_aws_ssm_with_instance_id=None):
+def get_default_ssh_options(environment, aws_ssm_target=None):
     default_ssh_options = []
 
-    strict_host_key_checking = environment.public_vars.get('commcare_cloud_strict_host_key_checking', True)
-    if not strict_host_key_checking or use_aws_ssm_with_instance_id:
-        default_ssh_options.append(('StrictHostKeyChecking', 'no'))
-
     known_hosts_filepath = environment.paths.known_hosts
-    if os.path.exists(known_hosts_filepath) and not use_aws_ssm_with_instance_id:
+    if os.path.exists(known_hosts_filepath):
         default_ssh_options.append(('UserKnownHostsFile', known_hosts_filepath))
-
-    if use_aws_ssm_with_instance_id:
-        default_ssh_options.append(('ProxyCommand', 'sh -c "aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters portNumber=%p"'))
-        default_ssh_options.append(('HostName', use_aws_ssm_with_instance_id))
+    else:
+        strict_host_key_checking = environment.public_vars.get('commcare_cloud_strict_host_key_checking', True)
+        if not strict_host_key_checking:
+            assert not aws_ssm_target, 'strict host key checking is required in AWS SSM environments'
+            default_ssh_options.append(('StrictHostKeyChecking', 'no'))
+
+    if aws_ssm_target:
+        sso_command = (
+            # NOTE {aws_ssm_target} can be changed to %h if the ssh
+            # hostname is aws_ssm_target (rather than an IP address)
+            f'aws ssm start-session --target {aws_ssm_target} '
+            '--document-name AWS-StartSSHSession '
+            '--parameters portNumber=%p'
+        )
+        assert '"' not in sso_command, """
+            Use {shlex_quote(sso_command)} and remove quotes (") in the
+            ProxyCommand below if it becomes necessary to put quotation
+            marks in sso_command. Otherwise it is more readable as is."""
+        default_ssh_options.append(('ProxyCommand', f'sh -c "{sso_command}"'))
```

Comment:
Is there a reason why we have to use this `sh -c "..."` wrapper for the proxy command? Would it work just as well to do this?
```suggestion
        default_ssh_options.append(('ProxyCommand', sso_command))
```

Reply:
I believe this comes straight from the AWS documentation https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-getting-started-enable-ssh-connections.html:

```
# SSH over Session Manager
host i-* mi-*
    ProxyCommand sh -c "aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters 'portNumber=%p'"
```

so that's where I got it from initially. I'm not sure whether it can be removed or not and why they chose to put it in there.

Reply:
I noticed that it was in AWS docs as well, but I'd like to know why it's there and if it's required?

Reply:
The command (`aws sso ...`) doesn't use any shell-specific functionality as we're using it here. I have seen examples of it used like `ProxyCommand sh -c 'AWS_PROFILE=<profile>:session aws ssm start-...'` where the shell is needed for the environment it provides.  As long as all of this is setup as desired by the thing calling `ssh`, I would assume removing the sh "wrapper" should be fine.

Reply:
Removed. We can always add it back if needed.